crates/orchestrator/src/jobs/types.rs

```diff
@@ -138,6 +140,57 @@ pub struct JobItem {
     pub updated_at: DateTime<Utc>,
 }
 
+/// Defining a structure that contains the changes to be made in the job object,
+/// id and created at are not allowed to be changed
+// version and updated_at will always be updated when this object updates the job
+#[derive(Serialize, Debug)]
+pub struct JobItemUpdates {
+    pub internal_id: Option<String>,
+    pub job_type: Option<JobType>,
+    pub status: Option<JobStatus>,
+    pub external_id: Option<ExternalId>,
+    pub metadata: Option<HashMap<String, String>>,
+}
+
+/// implements default selections for JobItemUpdates
+impl Default for JobItemUpdates {
+    fn default() -> Self {
+        JobItemUpdates { internal_id: None, job_type: None, status: None, external_id: None, metadata: None }
+    }
+}
+
+/// implements only needed singular changes
+impl JobItemUpdates {
+    pub fn only_status(status: JobStatus) -> Self {
+        JobItemUpdates { status: Some(status), ..Default::default() }
+    }
+    pub fn only_metadata(metadata: HashMap<String, String>) -> Self {
+        JobItemUpdates { metadata: Some(metadata), ..Default::default() }
+    }
+    pub fn to_document(&self, current_job: &JobItem) -> Result<Document> {
+        let mut doc = Document::new();
+
+        // Serialize the struct to BSON
+        let bson = bson::to_bson(&self)?;
+
+        // If serialization was successful and it's a document
+        if let Bson::Document(bson_doc) = bson {
+            // Add non-null fields to our document
+            for (key, value) in bson_doc.iter() {
+                if !matches!(value, Bson::Null) {
```

Comment:
if everything is null should we also throw an error with nothing to update maybe?

Reply:
While writing this code I assumed such case would not happen, lemme introduce a flag that checks if the not null case was ever executed, this will let us know if anything is there to be updated or not.

Reply:
[Here is the impl](https://github.com/madara-alliance/madara-orchestrator/pull/134/commits/a3996b4ec9f290724394ccaeffe17846c1003c9c).

Reply:
no we can skip that